narwhals/_pandas_like/group_by.py

```diff
@@ -37,46 +38,54 @@
 }
 
 
+def _group_dataframe(
+    df: PandasLikeDataFrame, keys: list[str], *, drop_null_keys: bool, reset_index: bool
+) -> pd.DataFrameGroupBy:
+    min_version = (1, 1)
+    if (
+        df._implementation is Implementation.PANDAS and df._backend_version < min_version
+    ):  # pragma: no cover
+        if (
+            not drop_null_keys
+            and df.simple_select(*keys)._native_frame.isna().any().any()
+        ):
```

Comment:
Unrelated from the PR itself but in a way I am not the biggest fan of this behaviour - this would raise only if there are actually null keys  🤔

Reply:
True, it is value-dependent, but it only affects old pandas versions, and null values in the column you're grouping by are hopefully not-too-common